core/database/migration/migration.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package migration
+
+import (
+	"context"
+
+	"github.com/juju/description/v4"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/database"
+)
+
+// Operation is a single step in a migration.
+type Operation interface {
+	// Setup is called before the operation is executed. It should return an
+	// error if the operation cannot be performed.
+	Setup(database.DBGetter) error
+
+	// Execute is called to perform the operation. It should return an error
+	// if the operation fails.
+	Execute(context.Context, description.Model) error
```

Comment:
how could we use this method for exports (is a signature like this correct)? (the idea is to have agnostic operations whether its export/import right?)

Reply:
The description model is mutable.